src/Architecture/Architecture.php

```diff
@@ -0,0 +1,112 @@
+<?php
+declare(strict_types=1);
+
+namespace Arkitect\Architecture;
+
+use Arkitect\Architecture\DSL\Component\Component;
+use Arkitect\Architecture\DSL\Component\DefinedBy;
+use Arkitect\Architecture\DSL\Component\MayDependOnAnyComponent;
+use Arkitect\Architecture\DSL\Component\MayDependOnComponents;
+use Arkitect\Architecture\DSL\Component\Rules;
+use Arkitect\Architecture\DSL\Component\ShouldNotDependOnAnyComponent;
+use Arkitect\Architecture\DSL\Component\Where;
+use Arkitect\Architecture\DSL\Layered\Layer;
+use Arkitect\Architecture\DSL\Modular\Module;
+use Arkitect\Expression\ForClasses\NotDependsOnTheseNamespaces;
+use Arkitect\Expression\ForClasses\ResideInOneOfTheseNamespaces;
+use Arkitect\Rules\Rule;
+
+class Architecture implements Component, DefinedBy, Where, MayDependOnComponents, MayDependOnAnyComponent, ShouldNotDependOnAnyComponent, Rules
+{
+    /** @var string */
+    private $componentName;
+    /** @var array<string, string> */
+    private $componentSelectors;
+    /** @var array<string, string[]> */
+    private $allowedDependencies;
+
+    private function __construct()
+    {
+    }
+
+    public static function withModules(): Module
```

Comment:
I am not sure about adding Modules, Layers and Components...
It's a semantic difference, but it adds a lot of code, and more things to explain to someone that reads an Arkitect configuration..

Reply:
I don't have a strong opinion about it. As for me we can remove _layers_ and _modules_ and leave the more generic _components_. I would like to have some feedback from @micheleorselli  and @AlessandroMinoccheri  as well.